runner/clients/proxy/proxy.go

```diff
@@ -56,6 +63,14 @@ func (p *ProxyServer) Run(ctx context.Context) error {
 	return nil
 }
 
+func (p *ProxyServer) PendingTxs() []*ethTypes.Transaction {
+	return p.pendingTxs
+}
+
+func (p *ProxyServer) ClearPendingTxs() {
+	p.pendingTxs = make([]*ethTypes.Transaction, 0)
+}
+
```

Comment:
wdyt about passing in the mempool to this proxy server and adding transactions directly? Then the mempool can handle nonce requests too (if it has a transaction in the mempool).

Feel free to remove the block splitting and gas limit from the mempool and just do a simple list like this that's returned whenever NextBlock is called.